### ISSUE TITLE
Strange visualization of the log during the test

### ISSUE THREAD

**Issue by utrobinmv** [PyTorchLightning/pytorch-lightning#8930]
## 🐛 Bug

Hello, 

in module 'test_epoch_end' I calculate the final metric

```py
score = self.metric(valid_preds, valid_labels)

print(score)

self.log("Final_metric", score)

logging_info('Test end Ok!')
```

but after getting the output, I found that the accuracy of the data written to the log differs from the original value

## Result

console output:
```
0.8403041825095058
Test end Ok!

DATALOADER:0 TEST RESULTS
{'Final_metric': 0.8403041958808899}
```

score 0.8403041825095058 != 0.8403041958808899

I think I can draw the wrong conclusion from such an error in the results?

pytorch-lightning==1.4.1
torch==1.9.0

**Comment by tchaton** [comment]
Dear @utrobinmv,

Would you mind opening providing a reproducible script using the boring_model: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report_model.py

Best,
T.C

**Comment by qqueing** [comment]
this is similar : https://github.com/PyTorchLightning/pytorch-lightning/issues/8887

**Comment by SkafteNicki** [comment]
This is due to the precision in float32 and vs float64:
```python
print(torch.tensor([0.8403041825095058], dtype=torch.float32).item())
# 0.8403041958808899

print(torch.tensor([0.8403041825095058], dtype=torch.float64).item())
# 0.8403041825095058
```

**Comment by tchaton** [comment]
I believe @SkafteNicki provided an answer and unfortunately, there is nothing we can do unless recommending you to use `Trainer(precision=64)`.

Closing this issue.
